client/rest/src/main/java/org/elasticsearch/client/DeadHostState.java

```diff
@@ -26,19 +26,23 @@
  * when the host should be retried (based on number of previous failed attempts).
  * Class is immutable, a new copy of it should be created each time the state has to be changed.
  */
-final class DeadHostState {
+final class DeadHostState implements Comparable<DeadHostState> {
 
     private static final long MIN_CONNECTION_TIMEOUT_NANOS = TimeUnit.MINUTES.toNanos(1);
     private static final long MAX_CONNECTION_TIMEOUT_NANOS = TimeUnit.MINUTES.toNanos(30);
 
-    static final DeadHostState INITIAL_DEAD_STATE = new DeadHostState();
-
     private final int failedAttempts;
     private final long deadUntilNanos;
+    private final TimeSupplier timeSupplier;
+
+    DeadHostState() {
+        this(TimeSupplier.DEFAULT);
+    }
 
-    private DeadHostState() {
+    DeadHostState(TimeSupplier timeSupplier) {
```

Comment:
I think it is worth javadocs on this to communicate that it is for building the *initial* dead state.